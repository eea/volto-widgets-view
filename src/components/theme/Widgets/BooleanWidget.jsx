import React from 'react';
import cx from 'classnames';

export const BooleanWidget = ({ children, className }) => (
  <span className={cx(className, 'boolean', 'widget')}>
    {children ? 'Yes' : 'No'}
  </span>
);
