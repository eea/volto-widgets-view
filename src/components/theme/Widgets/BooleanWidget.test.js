import React from 'react';
import renderer from 'react-test-renderer';
import { BooleanWidget } from './BooleanWidget';

describe('BooleanWidget', () => {
  it('renders a boolean true view widget component', () => {
    const component = renderer.create(
      <BooleanWidget className="metadata">{true}</BooleanWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a boolean false view widget component', () => {
    const component = renderer.create(
      <BooleanWidget className="metadata">{false}</BooleanWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
