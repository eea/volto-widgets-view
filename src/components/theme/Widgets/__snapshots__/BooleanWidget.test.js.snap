// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanWidget renders a boolean false view widget component 1`] = `
<span
  className="metadata boolean widget"
>
  No
</span>
`;

exports[`BooleanWidget renders a boolean true view widget component 1`] = `
<span
  className="metadata boolean widget"
>
  Yes
</span>
`;

exports[`BooleanWidget renders an empty boolean true view widget component 1`] = `
<span
  className="boolean widget"
>
  No
</span>
`;
